### ISSUE TITLE
Glcd not working.....

### ISSUE THREAD

**Issue by amjadali56** [positron96/ESP32_CNC_Pendant#4]
First of all Great Project and work.... Thanks a lot for such good job and sharing.
I'm using Devkit V1 !  No matter what I do, stuck on no-display LCD.... I'm using ST7902, tried both HW-SPI and SW-SPI but no luck... 
LCD is driven by 5V also the LCD-IO pins are connect directly without any extra components.  
Only LCD is connected to board, SD, POT and BTN are not connected.
No Change in Code. Except "_platform = espressif32@3.5.0"_  and for SW-SPI _"U8G2_ST7920_128X64_F_SW_SPI u8g2_{U8G2_R3, 16,  2, 15,  22}; "_
Checked the LCD and board with both SW-SPI and HW SPI example code which worked fine....

**Comment by positron96** [comment]
Hi. I can provide limited advice only, since I do not have access to your hardware.

1. Add debug UART output to the code and see what it does and what is does not. Maybe it hangs somewhere at the start.
2. Gradually transform example code that works into project code and see where it stops working. Or the other way around. 

Off the top of my head I can suggest looking at display bus speed (it's set to 600000 in main.cpp)

**Comment by amjadali56** [comment]
Wooww! Not expected such quick replay!  
Actually, this is my first time with VS code. 
Did UART output.... everything seems ok!
Bus speed is already set to 600000 in original code. with Rest pin delay of 100.
Yep! I think the only way is to gradually transform the code.
I do understand that nothing you can do without looking at my HW setup.... But you can Imagin my desperation.....  
Thanks again for such good work and your positive, well-timed reply.... Much obliged.  👍

**Comment by amjadali56** [comment]
I can print Hello Word! with HW and SW SPI.......
 ```
u8g2_.begin();
    //u8g2_.setBusClock(600000);
    u8g2_.setFont(u8g2_font_5x8_tr);
    u8g2_.setFontPosTop();
    u8g2_.setFontMode(1);
  
    u8g2_.drawStr(0,10,"Hello World!");	// write something to the internal memory
    u8g2_.sendBuffer();					// transfer internal memory to the display
    delay(10000);  

    //digitalWrite(PIN_RST_LCD, LOW);
    //delay(100);
    //digitalWrite(PIN_RST_LCD, HIGH);
```

**Comment by amjadali56** [comment]
OK.... Problem solved :)  The issue is in the Espressif@^3.5.0  library esp32-hal-spi.c file where there is an incompatible pointer.
...../platformio/packages/framework-arduinoespressif32@3.10006.210326/cores/esp32/esp32-hal-spi.c
The ‘result’ and ‘last_data’ are initialized as uint32_t.

The original code:
uint8_t * last_out8 = &result[c_longs-1];
uint8_t * last_data8 = &last_data;

To this:
uint32_t * last_out8 = &result[c_longs-1];
uint32_t * last_data8 = &last_data;

I made this change and it worked for me. Hope it help someone else..... Cheers and many many thanks for such a good project....

**Comment by positron96** [comment]
I'm very glad that you've fixed it! On a side note, I probably need to pin the version of ESP SDK to one that worked for me.
